library/core/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -1931,4 +1947,68 @@ private static boolean parseMsAcmCodecPrivate(ParsableByteArray buffer) throws P
 
   }
 
+  private class TrueHDSampleHolder {
+
+    private int counter;
+    private int samplesSize;
+    private long timeUs;
+    private @C.BufferFlags int blockFlags;
+    private boolean gotSyncFrame;
+    private boolean letFirstIn;
+    private Track track;
+
+    TrueHDSampleHolder() {
+      reset();
+    }
+
+    private boolean isFirstSampleCheck() {
+      return counter == 1;
+    }
+
+    private void resetCounters() {
+      track = null;
+      counter = 0;
+      samplesSize = 0;
+      timeUs = 0;
+      blockFlags = 0;
+    }
+
+    void reset() {
+      gotSyncFrame = false;
+      letFirstIn = true;
+      resetCounters();
+    }
+
+    void testSample(ExtractorInput input, int size, @C.BufferFlags int blockFlags, Track track)
+            throws IOException, InterruptedException {
+      samplesSize += size;
+      ++counter;
+      if (isFirstSampleCheck())
+        this.blockFlags = blockFlags;
+      if (!gotSyncFrame) {
+        byte[] bytes = new byte[12];
+        input.peekFully(bytes, 0, 12);
+        input.resetPeekPosition();
+        if (0 < Ac3Util.parseTrueHDSyncframeAudioSampleCount(ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN))) {
+          gotSyncFrame = true;
+          letFirstIn = false;
+        }
+      }
+      this.track = track;
+    }
+
+    boolean commit(long timeUs, boolean forceCommit) {
+      if (isFirstSampleCheck())
+        this.timeUs = timeUs;
+
+      boolean commit = counter == Ac3Util.TRUEHD_SAMPLE_COMMIT_COUNT || letFirstIn || forceCommit;
```

Comment:
I'm confused about letFirstIn -- why does the first sample we read not need to get batched into a group of TRUEHD_SAMPLE_COMMIT_COUNT samples?

I thought the way this should work is that each sample that is eventually output by this extractor starts with a syncframe and has TRUEHD_SAMPLE_COMMIT_COUNT samples from the container concatenated together (except for the last sample, which may be truncated). This logic doesn't appear to wait for a syncframe before starting output.

Reply:
Thanks, I'm taking it out.  Playback and seek seem fine without it now.  At one point it made things better in an older implementation, in an older exoplayer version, in an older version of ShieldTV firmware.  I know I made a fix after I had that in at one point to clean up some more seek issues and those may have fixed that as well.